File: Java/Medium/Gas-Station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int totalGas = 0;
        int totalCost = 0;
        int answer = 0;
        int remainingGas = 0;
        for (int i = 0; i<gas.length; i++) {
            totalGas += gas[i];
            totalCost += cost[i];
            remainingGas += (gas[i]-cost[i]);
            if (remainingGas < 0) {
                remainingGas = 0;
                answer = i+1;
            }
        }
        if (totalCost > totalGas) {
            return -1;
        }
        return answer;
    }
}

File: Python/Medium/Gas-Station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        n = len(gas)
        gasSum = 0
        costSum = 0
        gasTank = 0
        start = 0
        for i in range(n):
            gasSum+=gas[i]
            costSum+=cost[i]
            gasTank += gas[i]-cost[i]
            if gasTank < 0:
                gasTank = 0
                start = i+1
        if gasSum<costSum:
            return -1
        return start
